crates/jxl-render/src/features/spline.rs

```diff
@@ -261,12 +264,20 @@ pub fn render_spline(
                 (arc.point.y + max_distance + 1.5).floor() as i32,
             );
 
-            #[allow(clippy::needless_range_loop)]
-            for channel in 0..3 {
-                let buffer = &mut base_grid[channel];
+            for (channel, buffer) in base_grid.buffer_mut()[..3].iter_mut().enumerate() {
                 for y in ybegin..yend {
+                    let fy = y - region.top;
+                    if fy < 0 {
+                        continue;
+                    }
+
                     for x in xbegin..xend {
-                        let sample = buffer.get_mut(x as usize, y as usize).unwrap();
+                        let fx = x - region.left;
+                        if fx < 0 {
+                            continue;
+                        }
+
+                        let sample = buffer.get_mut(fx as usize, fy as usize).unwrap();
```

Comment:
```log
'main' panicked at 'called `Option::unwrap()` on a `None` value', crates/jxl-render/src/features/spline.rs:280:79
```
With `--crop` on images with splines

Example: jxl-dec ![spline jxl](https://github.com/tirr-c/jxl-oxide/assets/69643766/b123de74-625c-43e5-bf07-a0659a40144b) --crop "128 128 10 10"

Reply:
Oops, I'll take a look

Reply:
It was caused by checking only left bounds, but not right bounds. It is now fixed.

Reply:
Thanks!